### PR TITLE
[14849] Fixing CMakeLists.txt to avoid using curl (fails on windows installer ci

### PR DESCRIPTION
**Pull request by MiguelBarro**
Signed-off-by: Miguel Barro <miguelbarro@eprosima.com>

## Description

Windows installer CI fails because **cURL** cannot be invoke from **CMake** (no current explanation for this because it works fine locally ... note curl is distributed as an os tool from windows 10 henceforth).
The installer CI is running CMake 17 thus we can use the **CMake** `file(DOWNLOAD )` command instead of calling **cURL**.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [x] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [x] Documentation builds and tests pass locally. 
- [ ] NA New feature has been added to the `versions.md` file (if applicable).
- [ ] NA New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by MiguelBarro** [comment]
Successful generation of the [artifact](http://jenkins.eprosima.com:8080/job/FastRTPS%20Installer%20Windows/343/)

File: CMakeLists.txt [hunk 1]
```diff
@@ -405,12 +399,22 @@ if(BUILD_DOCUMENTATION)
 
     ### ReadTheDocs ########################
     if(NOT CHECK_DOCUMENTATION)
+
+        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/readthedocs_custom_template.cmake [=[
+
+            file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/v${PROJECT_VERSION}/htmlzip/" "eprosima-fast-rtps.zip")
+            # file(ARCHIVE_EXTRACT INPUT "eprosima-fast-rtps.zip" DESTINATION  "${PROJECT_BINARY_DIR}/doc/")
```
Comment (jsantiago-eProsima) [inline CMakeLists.txt:406]:
```suggestion
```

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (MiguelBarro) [reply CMakeLists.txt:407]: I have added a TODO explaining that this line should be upgraded with CMake 18

**Comment by MiguelCompany** [comment]
@mergifyio backport master

**Pull request merged**